src/test/groovy/com/vacation_bot/vacation/VacationServiceUnitTest.groovy

```diff
@@ -0,0 +1,54 @@
+package com.vacation_bot.vacation
+
+import com.vacation_bot.AbstractSpockUnitTest
+import com.vacation_bot.core.vacation.VacationService
+import com.vacation_bot.domain.models.UserModel
+import com.vacation_bot.domain.models.VacationTotalModel
+import com.vacation_bot.repositories.DefaultRepositoryFactory
+import com.vacation_bot.repositories.UserModelRepository
+import com.vacation_bot.repositories.VacationModelRepository
+import com.vacation_bot.repositories.VacationTotalRepository
+import com.vacation_bot.spring.exception.RepositoryException
+
+
+class VacationServiceUnitTest extends AbstractSpockUnitTest{
+
+    def 'exercise createVacation'() {
+        given: 'given valid subject under test'
+        def userModelRepository = Mock(UserModelRepository)
+        def vacationTotalRepository = Mock(VacationTotalRepository)
+        def vacationModelRepository = Mock(VacationModelRepository)
+
+        def factory = new DefaultRepositoryFactory([userModelRepository, vacationTotalRepository, vacationModelRepository])
+        def vacationService = new VacationService(factory)
+
+        and: 'valid input data'
+        def userName = 'Alex'
+        def startDate = '2017-10-02'
+        def endDate = '2017-10-20'
+
+        when: 'the exercised method is called when user is null'
+        vacationService.createVacation(userName, startDate, endDate)
+
+        then: 'repositories return expected exception'
+        1 * userModelRepository.findByNameOrAliases(userName, Arrays.asList(userName)) >> null
+
+        thrown(RepositoryException)
```

Comment:
It's better to write separate test for exercising exception because the exception is tested with each case from `where` block in the current implementation.